src/systems/user_commands/UserCommands.cc

```diff
@@ -424,9 +424,86 @@ bool CreateCommand::Execute()
     }
     case msgs::EntityFactory::kLight:
     {
-      // TODO(louise) Support light msg
-      ignerr << "light field not yet supported." << std::endl;
-      return false;
+      auto lightMsg = createMsg->light();
+      // Check if there's already a top-level entity with the given name
+      if (kNullEntity != this->iface->ecm->EntityByComponents(
+          components::Name(lightMsg.name()),
+          components::ParentEntity(lightMsg.parent_id())))
+      {
+        ignwarn << "Light entity named [" << lightMsg.name() << "] already"
+                << " exists. Entity not spawned."
+                << std::endl;
+        return false;
+      }
+
+      sdf::Light sdfLight = convert<sdf::Light>(lightMsg);
```

Comment:
Hum, I think this conversion doesn't make sense if we can't use `sdfLight` directly. We could construct the string below using the fields from the message, there's no need to convert to SDF first.

I now realize that there's no way to add a light to `sdf::Root` without converting it to string, and there's no way to get a  string if a light didn't come from an element (https://github.com/osrf/sdformat/issues/468) :confused: